File: Platforms/Mac/Native/Orbital.Host.Cocoa.Native/Host/Host/Display.m
#import "Display.h"

void Orbital_Host_Display_GetMainScreen(int* width, int* height)
{
    NSRect frame = [NSScreen.mainScreen frame];
    CGSize size = frame.size;
    *width = size.width;
    *height = size.height;
}

int Orbital_Host_Display_GetAllScreensCount(void)
{
    NSArray<NSScreen*>* screens = NSScreen.screens;
    return (int)screens.count;
}

void Orbital_Host_Display_GetAllScreens(int** widths, int** heights, int count)
{
    NSArray<NSScreen*>* screens = NSScreen.screens;
    NSUInteger screenCount = screens.count;
    if (count > screenCount) count = (int)screenCount;
    for (int i = 0; i < count; ++i)
    {
        NSRect frame = screens[i].frame;
        CGSize size = frame.size;
        *widths[i] = size.width;
        *heights[i] = size.height;
    }
}

File: Platforms/Mac/Shared/Orbital.Host.Mac/Display.cs
using System;
using System.Collections.Generic;
using AppKit;

namespace Orbital.Host.Mac
{
	public static class Displays
	{
		public static Display GetPrimaryDisplay()
		{
			var size = NSScreen.MainScreen.Frame.Size;
			return new Display(true, (int)size.Width, (int)size.Height);
		}

		public static Display[] GetDisplays()
		{
			var screens = NSScreen.Screens;
			var displays = new Display[screens.Length];
			for (int i = 0; i != displays.Length; ++i)
			{
				var size = screens[i].Frame.Size;
				displays[i] = new Display(i == 0, (int)size.Width, (int)size.Height);
			}
			return displays;
		}
	}
}
